minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -616,15 +630,21 @@ private static void addConnection(final Element parentElement, Map<String, Objec
             addTextElement(element, "labelIndex", "1");
             addTextElement(element, "zIndex", "0");
 
-            addTextElement(element, "sourceId", "Processor");
+            addTextElement(element, "sourceId", getValueString(connectionProperties, SOURCE_NAME_KEY));
```

Comment:
That is a very good way forward and I will refactor this to be two classes. I'll look in to Kwalify and google a bit to see what others are using for YAML schema validation.